### PR TITLE
NXP backend: Add support for AdaptiveAvgPool1d, fix AdaptiveAvgPool1d tests, add check for 4D pool ops inputs

### PR DESCRIPTION
**Pull request by roman-janik-nxp**
### Summary
Adds support for AdaptiveAvgPool1d, fixes tests for AdaptiveAvgPool1d. Adds check for 4D pool ops inputs. Requires PR NXP backend: Add pass for adding batch dimension for 3D input avgpool. Only last commit applies.

### Test plan
Unit tests provided.

cc @robert-kalmar

### CODE REVIEW

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/max_pool2d_with_indices_converter.py [hunk 1]
```diff
@@ -42,6 +42,10 @@ def _is_supported_in_IR(
         parameters_mapping: dict[str, Parameter],
         custom_delegation_options: CustomDelegationOptions,
     ) -> bool:
+        # # The input must be 4D.
```
Comment (novak-vaclav) [inline backends/nxp/backend/ir/converter/node_converters/ops_converters/max_pool2d_with_indices_converter.py]: I believe this should be uncommented, or removed.

File: backends/nxp/edge_passes/move_auxiliary_operator_into_separate_qdq_cluster_pass.py [hunk 2]
```diff
@@ -126,9 +123,15 @@ class MoveLeadingAuxiliaryOperatorIntoSeparateQDQClusterPass(NeutronEdgePass):
...
-        # MaxPool1D is represented in edge as Unsqueeze -> MaxPool2D -> Getitem -> Squeeze. The reshaping nodes must be moved out
+        # MaxPool1D is represented in edge as Unsqueeze -> MaxPool2D -> GetItem -> Squeeze. The reshaping nodes must be moved out
         #  of the cluster. Instead of [Un]squeeze, ViewCopy can be used as well.
-        MaxPool2D: [
+        MaxPool2DWithIndices: [
+            ViewCopy,
+            UnsqueezeCopy,
+        ],
+        # AdaptiveAvgPool1D is represented in edge as Unsqueeze -> AdaptiveAvgPool2D -> Squeeze. The reshaping nodes
+        # must be moved out of the cluster. Instead of [Un]squeeze, ViewCopy can be used as well.
```
Comment (novak-vaclav) [inline backends/nxp/edge_passes/move_auxiliary_operator_into_separate_qdq_cluster_pass.py:133]: nit: I think this part of the comment `ViewCopy can be used as well.` is not necessary, there are even more operators to use instead of `[Un]squeeze`.

File: backends/nxp/edge_passes/move_auxiliary_operator_into_separate_qdq_cluster_pass.py [hunk 3]
```diff
@@ -236,9 +239,15 @@ class MoveTrailingAuxiliaryOperatorIntoSeparateQDQClusterPass(NeutronEdgePass):
...
-        # MaxPool1D is represented in edge as Unsqueeze -> MaxPool2D -> Getitem -> Squeeze. The reshaping nodes must be moved out
+        # MaxPool1D is represented in edge as Unsqueeze -> MaxPool2D -> GetItem -> Squeeze. The reshaping nodes must be moved out
         #  of the cluster. Instead of [Un]squeeze, ViewCopy can be used as well.
-        Getitem: [
+        GetItem: [
+            ViewCopy,
+            SqueezeCopy,
+        ],
+        # AdaptiveAvgPool1D is represented in edge as Unsqueeze -> AdaptiveAvgPool2D -> Squeeze. The reshaping nodes
+        # must be moved out of the cluster. Instead of [Un]squeeze, ViewCopy can be used as well.
```
Comment (novak-vaclav) [inline backends/nxp/edge_passes/move_auxiliary_operator_into_separate_qdq_cluster_pass.py:249]: nit: I think this part of the comment `ViewCopy can be used as well.` is not necessary, there are even more operators to use instead of `[Un]squeeze`.

File: backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py [hunk 4]
```diff
@@ -66,32 +69,7 @@ def test__basic_nsys_inference(
             RandomDatasetCreator(low=-1, high=1),
             output_comparator=output_comparator,
             use_qat=use_qat,
-        )
-
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py:70]: Why was the test removed? If it works now, add it as a test case to `test__basic_nsys_inference` just in case it stops working again and mention in its `id` that it was a known bug and that is why it's kept.

File: backends/nxp/edge_passes/move_auxiliary_operator_into_separate_qdq_cluster_pass.py [hunk 5]
```diff
@@ -3,36 +3,35 @@
...
 
-import operator
-
 import torch
 
 from executorch.backends.nxp.edge_passes.neutron_edge_pass import NeutronEdgePass
 from executorch.backends.nxp.neutron_partitioner import QDQClusterRecognizer
-from executorch.backends.nxp.tests.ops_aliases import PermuteCopy
 
-# noinspection PyProtectedMember
```
Comment (novak-vaclav) [inline backends/nxp/edge_passes/move_auxiliary_operator_into_separate_qdq_cluster_pass.py:14]: Now that I realize, importing and using a piece of code from test module in a "production" code is wrong, ie. `executorch.backends.nxp.tests.ops_aliases` used in `executorch.backends.nxp.edge_passes`

This imo is a code smell and `ops_aliases` should be moved up one level to `executorch.backends.nxp` to remove the dependency.

It could also be moved to a global scope so that other backends can use it, however that would be up to debate with Meta and other contributors.

File: backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py [hunk 6]
```diff
@@ -140,3 +118,29 @@ def test__kernel_size_and_stride_limit_exceeded(self):
             delegated_ep.graph, [ExecutorchDelegateCall]
         )
         assert graph_contains_any_of_ops(delegated_ep.graph, [AdaptiveAvgPool2D])
+
+
+class TestAdaptiveAvgPool1D:
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py]: nit: imo the name of this class should be different since we are in `test_adaptive_avg_pool2d_converter.py`.
I suggest `TestAdaptiveAvgPool1DTo2D`

File: backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py [hunk 7]
```diff
@@ -140,3 +118,29 @@ def test__kernel_size_and_stride_limit_exceeded(self):
...
+        graph_verifier = DetailedGraphVerifier(
+            mocker,
+            expected_delegated_ops={AdaptiveAvgPool2D: 1, ViewCopy: 2},
+            expected_non_delegated_ops={},
+        )
+
+        remove_quant_io_ops = True  # Use quantized dataset.
+        output_comparator = AllCloseOutputComparator(atol=1)  # Allow single bit error.
+
+        lower_run_compare(
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py:144]: nit: please use the `RandomDatasetCreator(low=-1, high=1)` just like in other pooling tests

File: backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py [hunk 8]
```diff
@@ -140,3 +118,29 @@ def test__kernel_size_and_stride_limit_exceeded(self):
             delegated_ep.graph, [ExecutorchDelegateCall]
         )
         assert graph_contains_any_of_ops(delegated_ep.graph, [AdaptiveAvgPool2D])
+
+
+class TestAdaptiveAvgPool1D:
+
+    # Just a basic test to verify that the operator gets extended to the 2D variant correctly.
+    def test__basic_nsys_inference(self, mocker, request):
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py]: Please add a QAT test.

File: docs/source/backends/nxp/op-support.csv [hunk 9]
```diff
@@ -1,6 +1,7 @@
 Operator,Compute DType,Quantization,Constraints
 aten.abs.default,int8,static int8,
 aten._adaptive_avg_pool2d.default,int8,static int8,"ceil_mode=False, count_include_pad=False, divisor_override=False"
+aten.adaptive_avg_pool1d.default,int8,static int8,"ceil_mode=False, count_include_pad=False, divisor_override=False"
```
Comment (novak-vaclav) [inline docs/source/backends/nxp/op-support.csv]: Verify that this `"ceil_mode=False, count_include_pad=False, divisor_override=False"` is true for `adaptive_avg_pool`, but also for other pooling ops while you're at it.

I see some check in other pooling ops, but in `adaptive_avg_pool_2d_converter.py` I don't see anything.

File: docs/source/backends/nxp/op-support.csv [hunk 10]
```diff
@@ -1,6 +1,7 @@
 Operator,Compute DType,Quantization,Constraints
 aten.abs.default,int8,static int8,
 aten._adaptive_avg_pool2d.default,int8,static int8,"ceil_mode=False, count_include_pad=False, divisor_override=False"
```
Comment (novak-vaclav) [inline docs/source/backends/nxp/op-support.csv]: nit: now I realize that for `aten._adaptive_avg_pool2d.default`, there is an underscore prefix, but in `nxp-quantization.md`, there is not. Please normalize it.

**Review by novak-vaclav — COMMENTED** [review]
I've got a few comments, otherwise a good find and a good fix 👍😊

File: backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py (hunk 8 above)
Comment (roman-janik-nxp) [reply backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py]: Added also for Avgpool2d and Maxpool2d.

File: docs/source/backends/nxp/op-support.csv (hunk 10 above)
Comment (roman-janik-nxp) [reply docs/source/backends/nxp/op-support.csv]: With underscore is the Edge dialect, without the Aten dialect that is documented in `nxp-quantization.md`. So it's correct as it is.

File: backends/nxp/edge_passes/move_auxiliary_operator_into_separate_qdq_cluster_pass.py (hunk 5 above)
Comment (roman-janik-nxp) [reply backends/nxp/edge_passes/move_auxiliary_operator_into_separate_qdq_cluster_pass.py:14]: Yes, that's correct. I will move it `executorch.backends.nxp`.

File: backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py (hunk 8 above)
Comment (novak-vaclav) [reply backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py]: You meant Adaptive Avgpool2d, not just plain Avgpool2d, right?

**Review by novak-vaclav — APPROVED** [review]
I have an additional question to one of your answers. If it just happens to be a typo in your answer, then LGTM. Otherwise please fix it.
After that, we can merge

File: backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py (hunk 6 above)
Comment (roman-janik-nxp) [reply backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py]: Changed also for Avgpool2d and Maxpool2d.

File: backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py (hunk 8 above)
Comment (roman-janik-nxp) [reply backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py]: I meant plain Avgpool2d. That means for Adaptive Avgpool2d, Avgpool2d and Maxpool2d. But I forgot to do it for Avgpool2d, so I added it now.
Same for the rename `TestAdaptiveAvgPool1DTo2D`.

Comment (novak-vaclav) [reply backends/nxp/tests/ir/converter/node_converter/test_adaptive_avg_pool2d_converter.py]: Okey then fine by me, we can merge 😊
